### DISCUSSION TITLE
Can filament support 3D Gaussian Splatting(3DGS)?

### DISCUSSION THREAD

**Discussion by linghugoogle** [google/filament#8033] [General]
3DGS: https://repo-sam.inria.fr/fungraph/3d-gaussian-splatting/

**Comment by pixelflinger** [comment]
It can be implemented using our material system. It's not supported natively.

**Reply by PsycoTodd** [reply]
Hello, 

I think about this too. But there is a radixsort part normally implemented in compute shader. Is there a way to write compute shader in the current filament material system? Any example to follow? Thank you.

**Comment by rawnsley** [comment]
I'm also interested in using NeRFs and 3DGS within Google Filament applications (on Android in my case). There are a number of well-regarded open source implementations available now, such as [Spark](https://github.com/sparkjsdev/spark), which could provide a template or at least give an indication of complexity.

Any pointers would be gratefully received and I would suggest this is going to be a must-have feature of rendering engines going forward ([Google Maps appears to be all in](https://www.youtube.com/watch?v=gXug7Kb3p4I) for example).

**Comment by nmfisher** [comment]
@rawnsley I've glanced at this briefly before, the blocker seems to be what @PsycoTodd noted. Filament doesn't support compute shaders so there's no (easy) way to depth-sort the splats based on camera orientation on the GPU. You'd need to do this on the CPU side. In theory I suppose you could some kind of ping-pong texture sorting? Not sure how efficient that would be. Also off the top of my head I'm not sure if Filament supports fixed draw order for that many primitives.
